### ISSUE TITLE
Add training code

### ISSUE THREAD

**Issue by Abhigupta4981** [JoYoungjoo/SC-FEGAN#21]
Training code is necessary to understand what loss functions were used, what was the model architecture etc. It would be a nice idea to add training code too.

**Comment by abtExp** [comment]
I'm trying to implement it in keras. Training code would be helpful.

**Comment by Zealoe** [comment]
when will the training codes be released? Thanks.

**Comment by abtExp** [comment]
i mostly have completed the code with the loss functions and the model, but i'm facing some issues with the model architecture. If anyone can help i'll be able to completely reproduce the model in keras.
I've already created a question on keras google group and stack overflow, but still no luck.
If anyone of you have any ideas regarding my specific issue, feel free to answer.
Here's the link to the question:
<a href='https://stackoverflow.com/questions/55115527/invalidargumenterror-conv2dcustombackpropinput-input-and-out-backprop-must-hav'>StackOverflow</a>

**Comment by claudetheboof** [comment]
Man we really need a training code. Or at least a very well trained model. Buy training script would be super awesome!

**Comment by 594cp** [comment]
@abtE Thanks for your trying on the training code, do you have any solutions now?

**Comment by abtExp** [comment]
i'll update the repository this weekend. i still haven't recieved any response on the question, but i'll try to figure it out and will be working on it this weekend. Hopefully i'll get it.
🤞

**Comment by claudetheboof** [comment]
ugh still no luck in training codes?

**Comment by Janekaka** [comment]
@abtExp Hi! First of all, thank you for your generosity. May I ask you about your recent progress?

**Comment by abtExp** [comment]
@claudetheboof @Janekaka and everyone else, sorry for the delay. I've been very busy recently working on my final project submission for my graduation degree and simultaneously working on 2 major projects in my internship. I'll try my best to finish this this month, in the mean time, if anyone of you guys want, you can go ahead, fork my repo or submit a merge request if you can fix some thing or can implement something. Sorry again for the delay and thanks for your patience.

**Comment by Janekaka** [comment]
Sorry to bother you and thanks for your quick reply. Wish you the best.

**Comment by abtExp** [comment]
Hi all, finally got some time to work on the implementation, i've fixed the loss functions and the gated convolutions, also the data loader uses the masking technique mentioned in the paper. Although i haven't tested it much, but i think i'll soon complete it and train it. Check out the code at [my repo](https://github.com/abtExp/ArxivImpl/tree/master/1902.06838)

**Comment by muxgt** [comment]
@abtExp have you deleted your repository? the link is not available and I haven't found it in the list of your repositories. Any chance to see your code?

**Comment by abtExp** [comment]
@muxgt hi, sorry for that, I've changed the repository name and structure, I'm developing it now as a library. You can find the repository at : https://github.com/abtExp/arxivr

**Comment by muxgt** [comment]
@abtExp thanks! I will check it out. Did you receive good results with training?

**Comment by PngYng** [comment]
> @muxgt嗨，很抱歉，我已经更改了存储库名称和结构，现在正在将其开发为一个库。您可以在以下位置找到存储库：[https](https://github.com/abtExp/arxivr) : [//github.com/abtExp/arxivr](https://github.com/abtExp/arxivr)

hi，can you share the train code with me, thanks
